flair/embeddings.py

```diff
@@ -994,6 +1418,140 @@ def __str__(self):
         return self.name
 
 
+class OpenAIGPT2Embeddings(TokenEmbeddings):
+    def __init__(
+        self,
+        model: str = "gpt2-medium",
+        layers: str = "1",
+        pooling_operation: str = "first_last",
+        use_scalar_mix: bool = False,
+    ):
+        """OpenAI GPT-2 embeddings, as proposed in Radford et al. 2019.
+        :param model: name of OpenAI GPT-2 model
+        :param layers: comma-separated list of layers
+        :param pooling_operation: defines pooling operation for subwords
+        :param use_scalar_mix: defines the usage of scalar mix for specified layer(s)
+        """
+        super().__init__()
+
+        if model not in OPENAI_GPT2_PRETRAINED_MODEL_ARCHIVE_MAP.keys():
+            raise ValueError("Provided OpenAI GPT-2 model is not available.")
+
+        self.tokenizer = GPT2Tokenizer.from_pretrained(model)
+        self.model = GPT2Model.from_pretrained(
+            pretrained_model_name_or_path=model, output_hidden_states=True
+        )
+        self.name = model
+        self.layers: List[int] = [int(layer) for layer in layers.split(",")]
+        self.pooling_operation = pooling_operation
+        self.use_scalar_mix = use_scalar_mix
+        self.static_embeddings = True
+
+        dummy_sentence: Sentence = Sentence()
+        dummy_sentence.add_token(Token("hello"))
+        embedded_dummy = self.embed(dummy_sentence)
+        self.__embedding_length: int = len(
+            embedded_dummy[0].get_token(1).get_embedding()
+        )
+
+    @property
+    def embedding_length(self) -> int:
+        return self.__embedding_length
+
+    def _add_embeddings_internal(self, sentences: List[Sentence]) -> List[Sentence]:
+        self.model.to(flair.device)
+        self.model.eval()
+
+        sentences = _get_transformer_sentence_embeddings(
+            sentences=sentences,
+            tokenizer=self.tokenizer,
+            model=self.model,
+            name=self.name,
+            layers=self.layers,
+            pooling_operation=self.pooling_operation,
+            use_scalar_mix=self.use_scalar_mix,
+            bos_token="<|endoftext|>",
+            eos_token="<|endoftext|>",
+        )
+
+        return sentences
+
+
+class RoBERTaEmbeddings(TokenEmbeddings):
+    def __init__(
+        self,
+        model: str = "roberta.large",
+        layers: str = "-1",
+        pooling_operation: str = "first",
+        use_scalar_mix: bool = False,
+    ):
+        """RoBERTa, as proposed by Liu et al. 2019.
+        :param model: name of RoBERTa model
+        :param layers: comma-separated list of layers
+        :param pooling_operation: defines pooling operation for subwords
+        :param use_scalar_mix: defines the usage of scalar mix for specified layer(s)
+        """
+        super().__init__()
+
+        try:
+            self.model = torch.hub.load("pytorch/fairseq", model)
+        except:
+            log_line(log)
```

Comment:
log_line needs to be imported, otherwise this fails.
```python
from flair.training_utils import log_line
```

Reply:
Fixed :) I think I have to change my PyCharm default theme...